org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/utils/LoggerTest.java

```diff
@@ -0,0 +1,166 @@
+package org.eclipse.lsp4xml.utils;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.FileHandler;
+import java.util.logging.Handler;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+import org.eclipse.lsp4j.InitializeParams;
+import org.eclipse.lsp4j.MessageActionItem;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+import org.eclipse.lsp4j.ShowMessageRequestParams;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.lsp4xml.XMLLanguageServer;
+import org.eclipse.lsp4xml.internal.parser.XMLParser;
+import org.eclipse.lsp4xml.model.XMLDocument;
+import org.eclipse.lsp4xml.services.XMLLanguageService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+/**
+ * LoggerTest
+ */
+public class LoggerTest {
+  private String path = "src/test/java/org/eclipse/lsp4xml/utils/logTestFiles/testLogFile.log";
+  private MockLanguageClient mockLanguageClient;
+  private File logFile;
+  private XMLLanguageService languageService;
+  private XMLDocument xmlDocument;
+
+  @Before
+  public void startup() {
+    deleteLogFile();
+    mockLanguageClient = createLanguageClient(MessageType.Warning, "Log Message");
+    InitializeParams params = createInitializationOptionsParams(path);
+    LogHelper.initializeRootLogger(mockLanguageClient, XMLLanguageServer.getInitializationOptions(params));
+    xmlDocument = XMLParser.getInstance().parse("<tag></tag>", "test:uri");
+    languageService = new XMLLanguageService();
+    languageService.findDocumentHighlights(xmlDocument, new Position(-1,-1));
+    logFile = new File(path);
+  }
+  @After
+  public void cleanUp() {
+    Handler[] handlers = Logger.getLogger("").getHandlers();
+    for (Handler h : handlers) {
+      h.close();
+    }
+    deleteLogFile();
+  }
+
+
+  @Test
+  public void testLogCreated() {
```

Comment:
you want to check calling this instance's logger should log messages to the file and sent them to the client